lib/reservation_generator_helper.rb

```diff
@@ -0,0 +1,96 @@
+module ReservationGeneratorHelper
+  require 'ffaker'
+  include ActiveSupport::Testing::TimeHelpers
+
+  def gen_res(random = false)
+    r = Reservation.new(status: 'reserved', reserver_id: User.all.sample.id,
+                        equipment_model: EquipmentModel.all.sample,
+                        notes: FFaker::HipsterIpsum.paragraph(2),
+                        start_date: Time.zone.today)
+    len = r.equipment_model.maximum_checkout_length
+    len -= random ? rand(len - 1) : 1
+    r.due_date = Time.zone.today + len.days
+    r
+  end
+
+  def make_checked_out(res, _ = false)
+    res.status = 'checked_out'
+    res.checked_out = res.start_date
+    res.equipment_item = res.equipment_model.equipment_items.all.sample
+    res.checkout_handler_id = User.where('role = ? OR role = ? OR role = ?',
+                                         'checkout', 'admin',
+                                         'superuser').all.sample.id
+  end
+
+  def make_returned(res, random = false)
+    make_past(res, random)
+    make_checked_out res
+    r_date = if res.due_date > Time.zone.today
+               random ? rand(res.start_date..Time.zone.today) : Time.zone.today
+             else
+               random ? rand(res.start_date..res.due_date) : res.due_date
+             end
+    check_in(res, r_date)
+  end
+
+  def make_overdue(res, random = false)
+    make_past(res, random, true)
+    make_checked_out res
+    res.overdue = true
+  end
+
+  def make_returned_overdue(res, random = false)
+    make_overdue(res, random)
+    return_date = random ? rand(res.due_date..Time.zone.today) : Time.zone.today
+    check_in(res, return_date)
+  end
+
+  def make_missed(res, random = false)
+    make_past(res, random)
+    res.status = 'missed'
+  end
+
+  def make_archived(res, random = false)
+    make_past(res, random) if random && rand < 0.5
```

Comment:
I thought that archived reservations were mostly ones that had some sort of problem, which isn't likely to arise if the reservation hasn't been checked out/in the past, but I suppose that I didn't really replicate that, either. 

Perhaps it should just set the status and ignore time? Actually generating a "realistic" archived reservation (like one where the item was lost) seems like it could be messy, and there isn't much of a point to manipulating dates if it isn't going to be the closest we can get to a "real" archived reservation.